src/uController.SourceGenerator/Reflection/RoslynType.cs

```diff
@@ -77,6 +80,10 @@ public override Type[] GetGenericArguments()
 
         public override Type GetGenericTypeDefinition()
         {
+            if (NamedTypeSymbol.NullableAnnotation == NullableAnnotation.Annotated)
+            {
+                return typeof(Nullable<>);
+            }
```

Comment:
This doesn't seem right. `Nullable<T>` is an actual type while nullable annotations are compile time magic.

Reply:
💡 Yeah, an example of where I got a little caught up trying to follow the reflection-style patterns in this API. I had this in so I could derive the unannotated state of nullable types for when they're passed into `typeof` but I realize that's not necessary.